### PR TITLE
feat(pharos-image-card): add overlay slot

### PR DESCRIPTION
**Pull request by chrisjbrown**
This change adds an event emit when the user hovers over or away from the image.
This allows the consumer to react to these mouseenter/mouseleave events

**This change:** (check at least one)

- [x] Adds a new feature
- [ ] Fixes a bug
- [ ] Improves maintainability
- [ ] Improves documentation
- [ ] Is a release activity

**Is this a breaking change?** (check one)

- [ ] Yes
- [x] No

**Is the:** (complete all)

- [x] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [x] Test suite(s) passing?
- [x] Code coverage maximal?
- [ ] Changeset added?

**What does this change address?**
A clear and concise description or a direct link to an issue [e.g. #15]
If adding a feature, a rationale for its addition goes here.

**How does this change work?**
A clear, detailed description of how this change addresses the issue.
This could be a bullet list if there are several moving parts.

**Additional context**
Add any other context here.

### CODE REVIEW

File: packages/pharos/src/components/image-card/pharos-image-card.ts [hunk 1]
```diff
@@ -161,17 +165,20 @@ export class PharosImageCard extends LitElement {
...
+        href="${this.link}"
+        label=${ifDefined(this.imageLinkLabel)}
+        subtle
+        flex
+        no-hover
+        @mouseenter=${this._handleImageHover}
+        @mouseleave=${this._handleImageHover}
+        >${this._renderLinkContent()}${this._renderHoverMetadata()}</pharos-link
+      >
+      ${this._imgHover ? html`<slot name="overlay"></slot>` : nothing}
```
[the remark this answers is not in the document]
Comment (chrisjbrown) [reply packages/pharos/src/components/image-card/pharos-image-card.ts:180]: 🤔 maybe 
one thing for my situation. 
I want to only hover when the `pharos-link` is hovered... and can't tell that without adding an eventlistener to the `pharos-link` element

[the remark this answers is not in the document]
Comment (daneah) [reply packages/pharos/src/components/image-card/pharos-image-card.ts:180]: Right—I think it makes sense to keep this model for the moment, then.

**Review by daneah — APPROVED** [review]

File: packages/pharos/src/components/image-card/pharos-image-card.ts [hunk 2]
```diff
@@ -127,6 +127,13 @@ export class PharosImageCard extends LitElement {
...
   private _handleImageHover(event: Event): void {
     this._title['_hover'] = event.type === 'mouseenter';
+
+    const mouseEvent = new CustomEvent(
+      event.type === 'mouseenter'
+        ? 'pharos-image-card-image-mouseenter'
+        : 'pharos-image-card-image-mouseleave'
+    );
+    this.dispatchEvent(mouseEvent);
   }
```
Comment (michael-iden) [inline packages/pharos/src/components/image-card/pharos-image-card.ts:137]: Not a blocker but I wonder if it makes sense to have 2 different handlers (one for mouse enter and other for mouse leave). Seems like that would simplify a couple conditionals without too much code duplication 🤷

**Review by michael-iden — APPROVED** [review]

File: packages/pharos/src/components/image-card/pharos-image-card.ts (hunk 2 above)
Comment (chrisjbrown) [reply packages/pharos/src/components/image-card/pharos-image-card.ts:137]: yeah i could see that going both ways. 
the ` _handleImageHover` event already existed for the title hover so i just followed that path 🤔

Comment (chrisjbrown) [reply packages/pharos/src/components/image-card/pharos-image-card.ts]: updated. i would say it looks cleaner

**Pull request merged**